### PR TITLE
Fixes #291 Model should be ableo to retrieve MolWeight parameters for…

### LINKED ISSUE #291: Model should be ableo to retrieve MolWeight parameters for parameter defined in a Molecule
see https://github.com/Open-Systems-Pharmacology/MoBi/issues/209

and 
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/f75c2b2f6a639a8a8f27e4937358ad7819e87ba0/src/OSPSuite.Core/Domain/Model.cs#L55

### PR DESCRIPTION
**Pull request by msevestre**
… parameter defined in a Molecule

### CODE REVIEW

File: src/OSPSuite.Core/Domain/Model.cs [hunk 1]
```diff
@@ -35,39 +37,32 @@ public IContainer Neighborhoods
...
+            Root?.Add(_neighborhoods);
          }
-         get { return _neighborhoods; }
+         get => _neighborhoods;
       }
 
       public double? MolWeightFor(IQuantity quantity)
       {
-         if (Root == null)
+         if (Root == null || quantity == null)
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Model.cs]: This method was too restrictive I believe. It was returning the Molweight of the associated quantity only for `MoleculeAmount` AND `Observer`. But it should be possible to retrieve the MolWeight for all other types of quantity (I think the last remaining one is `Parameter`, but no need to be explicit about that here)

**Comment by msevestre** [comment]
@Yuri05 review please

File: src/OSPSuite.Core/Domain/Model.cs [hunk 2]
```diff
@@ -35,39 +37,32 @@ public IContainer Neighborhoods
...
             return null;
 
          //Mol weight only available for molecule amount and observer
```
Comment (Yuri05) [inline src/OSPSuite.Core/Domain/Model.cs]: the comment is not valid anymore

**Comment by msevestre** [comment]
done

**Pull request merged**